sway/tree/view.c

```diff
@@ -244,25 +261,32 @@ void view_autoconfigure(struct sway_view *view) {
 	double y_offset = 0;
 
 	if (!container_is_floating(con) && ws) {
-
 		bool smart = config->hide_edge_borders_smart == ESMART_ON ||
 			(config->hide_edge_borders_smart == ESMART_NO_GAPS &&
 			!gaps_to_edge(view));
-		bool hide_smart = smart && view_ancestor_is_only_visible(view);
 
 		if (config->hide_edge_borders == E_BOTH
-				|| config->hide_edge_borders == E_VERTICAL || hide_smart) {
+				|| config->hide_edge_borders == E_VERTICAL) {
 			con->border_left = con->x != ws->x;
 			int right_x = con->x + con->width;
 			con->border_right = right_x != ws->x + ws->width;
 		}
+
 		if (config->hide_edge_borders == E_BOTH
-				|| config->hide_edge_borders == E_HORIZONTAL || hide_smart) {
+				|| config->hide_edge_borders == E_HORIZONTAL) {
 			con->border_top = con->y != ws->y;
 			int bottom_y = con->y + con->height;
 			con->border_bottom = bottom_y != ws->y + ws->height;
 		}
 
+		if (smart && view_ancestor_is_only_visible(view)) {
```

Comment:
I'm also not sure that both `view_ancestor_is_only_visible` and `view_is_only_visible_in_tab_or_stack` need to be called here. The two functions are only slightly different. It looks like the former should be used for smart gaps (#5407 - which also appears to be what caused this issue) and the latter for smart borders.

Reply:
Replacing the `view_ancestor_is_only_visible` with `view_is_only_visible_in_tab_or_stack` makes the top border still shown, since `con->y != ws->y` due to the tabs being in the way:

![image](https://user-images.githubusercontent.com/1403503/85910917-a2865200-b7ef-11ea-98be-0e808d17d5b1.png)

Reply:
That makes sense for the original code. It doesn't look like the call to `view_ancestor_is_only_visible` is needed with this iteration though. It doesn't hit the case above anyway.

Reply:
Thanks, confirmed that's the case and removed the extra call.